src/commands/handlers/registration.js

```diff
@@ -354,6 +356,29 @@ var handlers = {
     }
 };
 
+/**
+ * Only use the nick+password combo if an account has not been specifically given.
+ * If no account:{account,password} has been given, use the nick+password if available
+ */
+function getSaslAuth(handler) {
+    let options = handler.connection.options;
+    if (options.account && options.account.account) {
+        return {
+            account: options.account.account,
+            password: options.account.password || '',
```

Comment:
This PR looks fine, the only question I have is blank passwords. Does any auth flow support that?

Reply:
I'm not sure if any support it, but there's no harm in accepting a blank password while the account name must definitely not be empty.